srv/salt/ceph/mon/default.sls

```diff
@@ -33,17 +33,23 @@ create_mon_fs:
         - file: /var/lib/ceph/tmp/keyring.mon
 
 
-# TODO: we can put this check in a nice exec. module
-start-mon:
-  cmd.run:
-    - name: "systemctl start ceph-mon@{{ grains['host'] }}"
+mon-start:
+  service.running:
+    - name: ceph-mon@{{ grains['host'] }}
     - require:
-        - cmd: create_mon_fs
+      - cmd: create_mon_fs
+    - enable: True
+
 
-enable-mon:
+verify_mon_running:
   cmd.run:
-    - name: "systemctl enable ceph-mon@{{ grains['host'] }}"
     - require:
-        - cmd: create_mon_fs
+      - service: mon-start
+    - name: |
+        sleep 5
+        systemctl status ceph-mon@{{ grains['host'] }}
+        mon_status=$?
+        test $mon_status -eq 0 || echo "The ceph-mon@{{ grains['host'] }} unit failed to start"
+        test $mon_status -eq 0
 
 
```

Comment:
Rather than the hardcoded sleep, I think one of these will be better:

```
check mon:
  module.run:
    - name: cephprocesses.check
    - kwargs:
        'roles':
          - mon
    - fire_event: True
    - failhard: True
```

```

wait for mon:
  module.run:
    - name: cephprocesses.wait
    - kwargs:
        'timeout': 6
        'delay': 2
        'roles':
          - mon
    - fire_event: True
    - failhard: True
```

The only reason to  use the wait version is if we think these processes are slow to start.  But the original problem is saying that these start and fail immediately.  Then, the check will be enough.

Reply:
This would be the same for mgr as well.  We are using the examples as part of the restart logic.  Take a look at /srv/salt/ceph/processes/`mon|mgr`/default.sls.

Reply:
@swiftgist I made the change you suggested.